extensions/security-jpa/deployment/src/main/java/io/quarkus/security/jpa/deployment/QuarkusSecurityJpaProcessor.java

```diff
@@ -195,7 +201,11 @@ private void generateIdentityProvider(Index index, JpaSecurityDefinition jpaSecu
                     case CUSTOM:
                         passwordProviderClassStr = passwordProviderValue.asString();
                         passwordProviderMethod = "getPassword";
+                        objectToInvokeOn = methodCreator.readInstanceField(passwordProviderField, methodCreator.getThis());
+                        BytecodeCreator trueBranch = methodCreator.ifNull(objectToInvokeOn).trueBranch();
                         objectToInvokeOn = methodCreator.newInstance(MethodDescriptor.ofConstructor(passwordProviderClassStr));
+                        methodCreator.writeInstanceField(passwordProviderField, methodCreator.getThis(), objectToInvokeOn);
```

Comment:
I think you need to call `trueBranch.writeInstanceField` here, otherwise you will call it always. Also I'm not sure you can actually re-use `objectToInvokeOn`, but perhaps I'm wrong.

Reply:
Refactored with:

```java
ResultHandle passwordProviderInstanceField = methodCreator.readInstanceField(passwordProviderField, methodCreator.getThis());
BytecodeCreator trueBranch = methodCreator.ifNull(passwordProviderInstanceField).trueBranch();
ResultHandle passwordProviderInstance = trueBranch.newInstance(MethodDescriptor.ofConstructor(passwordProviderClassStr));
trueBranch.writeInstanceField(passwordProviderField, methodCreator.getThis(), passwordProviderInstance);
trueBranch.close();
objectToInvokeOn = methodCreator.readInstanceField(passwordProviderField, methodCreator.getThis());
```